system/src/Grav/Common/Page/Medium/ImageMedium.php

```diff
@@ -587,7 +597,9 @@ protected function saveImage()
             $this->image->merge(ImageFile::open($overlay));
         }
 
-        return $this->image->cacheFile($this->format, $this->quality);
+        $cachedPath = $this->image->cacheFile($this->format, $this->quality);
+        $this->set('filepath', $cachedPath);
```

Comment:
This line breaks image handling as the image quality gets worse and worse on every resize.